robot/src/main/java/us/ilite/robot/modules/HatchFlower.java

```diff
@@ -1,73 +1,353 @@
 package us.ilite.robot.modules;
 
+import java.util.Arrays;
+import java.util.List;
+
+import com.flybotix.hfr.codex.Codex;
 import com.flybotix.hfr.util.log.ILog;
 import com.flybotix.hfr.util.log.Logger;
 
 import edu.wpi.first.wpilibj.Solenoid;
+import us.ilite.common.config.DriveTeamInputMap;
 import us.ilite.common.config.SystemSettings;
+import us.ilite.common.types.input.ELogitech310;
+import us.ilite.robot.commands.CommandQueue;
+import us.ilite.robot.commands.Delay;
+import us.ilite.robot.commands.ICommand;
+import us.ilite.robot.commands.ParallelCommand;
 
-
+/**
+ * Control the Hatch Flower
+ * 
+ * The Hatch Capture button will take precedence over the Hatch Push button.
+ * Once the Hatch Capture button is pressed, the hatch flower will transition to
+ * the CAPTURE state and stay there until the Hatch Push button is pressed.
+ * Then the Hatch Flower will push the hatch and go to the RELEASE configuration.
+ */
 public class HatchFlower extends Module {
 
+    public enum HatchFlowerStates {
+        CAPTURE, 
+        PUSH,
+        RELEASE;
+    }
+
     private ILog mLog = Logger.createLog(HatchFlower.class);
 
     private Solenoid solenoidOpenClose;
     private Solenoid solenoidExtension;
 
-    public enum SoleniodStates {
-        // TODO hold solenoid states for each state
-        CAPTURE,
-        RELEASE,
-        PUSH;
+    // Needed for prototype testing
+    // private final Codex<Double, ELogitech310> mController;
+
+    private CommandQueue mCurrentCommandQueue;
+
```

Comment:
We'll always start the match with either a hatch or cargo.  So the initial state for the hatch will always be 'CAPTURE'.